### PR TITLE
fix(document): display stored value

### PR DESCRIPTION
**Pull request by Rom1-B**
## Checklist before requesting a review

*Please delete options that are not relevant.*

- [x] I have read the CONTRIBUTING document.
- [x] I have performed a self-review of my code.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] This change requires a documentation update.

## Description

- It fixes !34700
- Here is a brief description of what this PR does
The document dropdown failed to load with an initial value (either the default or a previously saved value). This issue occurs with the Fields plugin, which allows adding a document-type field that leverages the core's dropdown functionality.

### CODE REVIEW

File: ajax/dropdownRubDocument.php [hunk 1]
```diff
@@ -81,7 +81,8 @@
             'width'     => '50%',
             'entity'    => intval($_POST['entity']),
             'rand'      => intval($_POST['rand']),
-            'condition' => ['glpi_documents.documentcategories_id' => (int)$_POST["rubdoc"]]
+            'condition' => ['glpi_documents.documentcategories_id' => (int)$_POST["rubdoc"]],
+            'value'     => $_POST['value'] ?? -1,
```
Comment (trasher) [inline ajax/dropdownRubDocument.php:85]: Not sure raw value should be passed

Comment (trasher) [reply ajax/dropdownRubDocument.php:85]:
```suggestion
            'value'     => (int)($_POST['value'] ?? -1),
```

File: src/Document.php [hunk 2]
```diff
@@ -1732,7 +1739,8 @@ public static function dropdown($options = [])
         $out .= "<span id='show_" . $p['name'] . "$rand'>";
         $out .= "</span>\n";
 
-        $params['rubdoc'] = 0;
+        $params['rubdoc'] = $p['rubdoc'] ?? 0;
+        $params['value'] = $p['value'] ?? 0;
```
Comment (trasher) [inline src/Document.php:1743]: Defaiult values shouyld be declared at the top of the method, with other ones

Comment (Rom1-B) [reply src/Document.php:1743]: The `$params` are sent just before the call to `Ajax::updateItemOnSelectEvent()`, but they should not contain these values.

**Review by AdrienClairembault — APPROVED** [review]

**Review by cedric-anne — APPROVED** [review]

**Pull request merged**